include/igl/slice_mask.cpp

```diff
@@ -66,7 +66,11 @@ IGL_INLINE void igl::slice_mask(
         {
           if(R(i))
           {
-            Y.row(yi++) = X.row(i);
+            for(int j = 0;j<X.cols();j++)
+            {
+              Y(yi,j) = X(i,j);
```

Comment:
I believe this is getting around an annoying Eigen static assert. If `Y` is `Eigen::Matrix<double,Eigen::Dynamic,3>` then it will complain if `X` is `Eigen::Matrix<double,Eigen::Dynamic,Eigen::Dynamic>`. An alternative is `Y.row(yi++).head(X.cols())`.

Reply:
Ah yes. I'm not sure the `Y.row(yi++).head(X.cols())` workaround would work if Y has a fixed column size and X does not, so might as well write the loop.

Reply:
oops, it'd be the other way around `Y.row(yi++) = X.row(i).head(Y.cols())`